pulsar-testclient/src/main/java/org/apache/pulsar/testclient/PerformanceTransaction.java

```diff
@@ -0,0 +1,685 @@
+                        } catch (Exception e) {
+                            log.error("Failed to build Producer/Consumer with exception : " + e);
+                        }
+                        AtomicReference<Transaction> atomicReference = buildTransaction(client, arguments);
+                        //The while loop has no break, and finally ends the execution through the shutdownNow of
+                        //0the executorService
+                        while (true) {
+                            try {
+                                Transaction transaction = atomicReference.get();
+                                for (List<Consumer<byte[]>> subscriptions : consumers) {
+                                    while(true) {
+                                        if(messageReceived.sum() == arguments.numMessagesReceivedPerTransaction){
+                                            break;
+                                        }
+                                        for (Consumer<byte[]> consumer : subscriptions) {
+                                            if (messageReceived.sum() == arguments.numMessagesReceivedPerTransaction) {
+                                                break;
+                                            }
+                                            Message message = null;
+                                            try {
+                                                message = consumer.receive(2, TimeUnit.SECONDS);
+                                                log.info("Receive message {} ", message);
+                                            } catch (Exception e) {
+                                                log.error("{} can`t receive message in 2 sec with exception {}",
+                                                        consumer, e);
+                                            }
+
+                                            messageReceived.increment();
+                                            long receiveTime = System.nanoTime();
+                                            if (arguments.isEnableTransaction) {
+                                                consumer.acknowledgeAsync(message.getMessageId(), transaction)
+                                                        .thenRun(() -> {
+                                                            long latencyMicros = NANOSECONDS.toMicros(
+                                                                    System.nanoTime() - receiveTime);
+                                                            messageAckRecorder.recordValue(latencyMicros);
+                                                            messageAckCumulativeRecorder.recordValue(latencyMicros);
+                                                        }).exceptionally(exception -> {
+                                                    if (exception instanceof InterruptedException && !executing.get()) {
```

Comment:
Why should we handle this exception separately?

Reply:
when tasks are executed completely, `executorService.shutdownNow();` will be executed.  At this time, other threads may be executing, and an interrupt exception may occur.

Reply:
ok